pkgs/games/xash3d/wrapper.nix

```diff
@@ -0,0 +1,59 @@
+{ writeShellScriptBin
+, makeDesktopItem
+, symlinkJoin
+, hlsdk
+, xash3d-unwrapped
+, name # codename, normally matches gamedir
+, fullname # the human readable name (e.g. "Half-Life: Opposing Force")
+, gamedir ? name # the name of the directory containing game data (e.g. "gearbox")
+, basedir ? "$HOME/.local/share/xash3d"
+, dllname ? name # ARGH why couldn't this just be always name (e.g. "opfor")
+}:
+
+let
+  description = "${fullname} running using the Xash3D engine, a GoldSrc reimplementation";
+
+  wrapper = writeShellScriptBin "xash3d-${name}" ''
+    export XASH3D_EXTRAS_PAK1=${xash3d-unwrapped}/share/xash3d/valve/extras.pk3
```

Comment:
You should set `XASH3D_RODIR`:
```
export XASH3D_RODIR="@out@/lib/xash3d"
```

I suggest `substituting` this to `$out/bin/xash3d` at install-time.

But then again, I'm biased ;)

Reply:
Why setting it ? First of all `$XASH3D_RODIR` is the directory to the assets, so why should it be `@out@/lib/xash3d` ? When not specified it defaults to `$XASH3D_BASEDIR` which is exactly what we want.